### PR TITLE
Running gofmt -s -w on the source tree.

### PR DESCRIPTION
**Pull request by alainjobart**
Anybody has an opinion on whether we should run it as we run goimports before submit?

I'd say we should, code is simpler after a run.

### CODE REVIEW

**Comment by sougou** [comment]
I like the -s format. I try to use it on all my new code. So, +1 from me.

**Comment by guoliang100** [comment]
+1

**Comment by guoliang100** [comment]
LGTM

**Comment by michael-berlin** [comment]
LGTM

Including gofmt -s as check in our Git hooks sounds good to me. But in that case we should probably output a diff to better educate the user why their current version is not "gofmt -s" compliant.

(In Eclipse I have Ctrl + Shift + F set to run "goimports". So I would be confused if a "gofmt" fails on a file formatted by "goimports" :))

File: go/bufio2/bufio_test.go [hunk 1]
```diff
@@ -292,7 +292,7 @@ func TestUnreadRuneError(t *testing.T) {
 	if err != nil {
 		t.Error("unexpected error on ReadRune (2):", err)
 	}
-	for _ = range buf {
+	for range buf {
```
Comment (enisoc) [inline go/bufio2/bufio_test.go]: This syntactic form was added in Go 1.4. Are we ok with dropping support for 1.3 and below?

**Pull request merged**